src/components/ui/calculator-form.tsx

```diff
@@ -9,10 +9,34 @@ import { Select, SelectTrigger, SelectValue, SelectContent, SelectItem } from "@
 import { Button } from "@/components/ui/button";
 import { CalculateInput } from "@/schema/api";
 import { Location, PowerPlantClass } from "@/schema/egrid";
+import { z } from "zod";
 
 const powerPlantOptions = PowerPlantClass.options;
 const locationOptions = Location.options;
 
+const currentYear = new Date().getFullYear();
+
+export const CalculateInputSchema = z.object({
```

Comment:
This object looks very similar to the `CalculateInput` object in the schema. Could we add these validation rules to the schema instead? Then we can just reuse the `CalculateInput` object instead of creating a new one. Also it would be helpful to add some unit tests for these rules